### DISCUSSION TITLE
How to automatically trigger workflow once

### DISCUSSION THREAD

**Discussion by woola168** [elsa-workflows/elsa-core#2852] [Q&A]
Is there anyway to run workflow once automatically using Timer and block it from running again or is there another way to do it 

I want to trigger a every 5 minutes looping workflow but just trigger it once and without using endpoint for every time the service redeploy it will auto run ( my thought is that using timer activities will automatic run every time when the service is up but i just want to trigger it once)

**Comment by sfmskywalker** [comment]
You could try configuring your workflow to be **Singleton** (configurable through the designer and through the fluent API).
When a workflow is marked as a "Singleton", only one **active** instance of that workflow is allowed to exist, until it either completes, faults or gets cancelled.
